ngraph/core/reference/include/ngraph/runtime/reference/deformable_convolution.hpp

```diff
@@ -14,31 +14,54 @@ namespace ngraph
         {
             namespace def_conv_impl
             {
-                inline void validate_params(const Shape& in_shape,
-                                            const Shape& o_shape,
-                                            const Shape& f_shape,
-                                            const Strides& strides,
-                                            const Strides& dilations,
-                                            const CoordinateDiff& pads_begin,
-                                            const CoordinateDiff& pads_end)
+                inline void validate_deformable_convolution_params(const Shape& in_shape,
+                                                                   const Shape& o_shape,
+                                                                   const Shape& f_shape,
+                                                                   const Shape& out_shape,
+                                                                   const Strides& strides,
+                                                                   const Strides& dilations,
+                                                                   const CoordinateDiff& pads_begin,
+                                                                   const CoordinateDiff& pads_end,
+                                                                   const int64_t& groups,
+                                                                   const int64_t& deformable_groups)
                 {
                     // this implementation supports 2D deformable convolutions
                     NGRAPH_CHECK(in_shape.size() == 4, "Unsupported input rank: ", in_shape);
-
                     NGRAPH_CHECK(o_shape.size() == 4, "Unsupported offset rank: ", o_shape);
-
                     NGRAPH_CHECK(f_shape.size() == 4, "Unsupported kernel rank: ", f_shape);
 
-                    const auto spatial_dims = in_shape.size() - 2;
-                    NGRAPH_CHECK(strides.size() == spatial_dims,
-                                 "Strides not definied for all and only spatial dimensions");
+                    NGRAPH_CHECK(in_shape[1] % groups == 0,
```

Comment:
I'm not sure but may be should to add `ngraph_check` for `groups` and `deformable_groups` (should be greater than 1)?

Similarly for `dilations` and `pads_begin` , `pads_end` (greater than 0)

Reply:
Good point. So `groups` and `deformable_groups` are attributes of the op. There are two node validation [checks](https://github.com/openvinotoolkit/openvino/blob/9acace614d4d2ea62c201a8b5259ffd7ffca5fe4/ngraph/core/src/op/deformable_convolution.cpp#L96) taking care of this in `validate_and_infer_types`. The idea why I introduced these additional checks is because when having dynamic shapes, node validation checks related to shapes are by-passed, and later in reference incompatible shapes may be called.
I didn't introduced checks for attributes since in `evaluates_map` we use `op->get_group()` and `op->get_deformable_group()` to get these values and they are validates already.

As for `pads_begin` and `pads_end` they can be zero, which means no padding (or valid padding)

Reply:
Yes, I wrote the incorrect conditions (didn't include the values themselves, one and zero). Thanks!